hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/MasterProcedureScheduler.java

```diff
@@ -135,7 +139,11 @@ protected void enqueue(final Procedure proc, final boolean addFront) {
       doAdd(tableRunQueue, getTableQueue(getTableName(proc)), proc, addFront);
     } else if (isServerProcedure(proc)) {
       ServerProcedureInterface spi = (ServerProcedureInterface) proc;
-      doAdd(serverRunQueue, getServerQueue(spi.getServerName(), spi), proc, addFront);
+      if (spi.hasMetaTableRegion()) {
+        doAdd(serverHighPriorityRunQueue, getServerQueue(proc), proc, addFront);
```

Comment:
Does this make it so all procedures associated with this server are 'high' priority?

Reply:
No, only the server procedure

Reply:
IIRC, there is a problem that, the sub procedures will inherit some resources from the parent procedure, so changing the logic here is easy to introduce dead lock...